EventFilter/Utilities/plugins/GlobalEvFOutputModule.cc

```diff
@@ -429,17 +463,40 @@ namespace evf {
       edm::LuminosityBlockForOutput const& iLB) const {
     auto openDatFilePath = edm::Service<evf::EvFDaqDirector>()->getOpenDatFilePath(iLB.luminosityBlock(), streamLabel_);
 
-    return std::make_shared<GlobalEvFOutputEventWriter>(openDatFilePath, iLB.luminosityBlock());
+    auto ret = std::make_shared<GlobalEvFOutputEventWriter>(openDatFilePath, iLB.luminosityBlock(), metaDataCache_);
+    lastWriter_ = ret.get();
```

Comment:
Do we guarantee the `LuminosityBlockCache` of the last lumi for which `globalBeginLuminosityBlock()` is called is alive at the time the `respondToOpenInputFile()` is called? I'd guess we do, but just to make sure.

Reply:
File opens do not cause an end luminosity block nor end run transition on their own. Only after the file is open could we cause an end lumi or end run transition to occur. Therefore all lumi and run caches are fine.